stdlib/universe/window.go

```diff
@@ -46,80 +48,74 @@ func CreateWindowOpSpec(args flux.Arguments, a *flux.Administration) (flux.Opera
 
 	spec := new(WindowOpSpec)
 
-	every, everySet, err := func() (flux.Duration, bool, error) {
-		d, ok, err := args.GetDuration("every")
-		if err != nil || !ok {
-			return flux.Duration{}, ok, err
+	if every, err := func() (flux.Duration, error) {
+		d, err := args.GetRequiredDuration("every")
+		if err != nil {
+			return flux.Duration{}, err
 		}
 
 		if d.IsNegative() {
-			return flux.Duration{}, false, errors.New(codes.Invalid, `parameter "every" must be non-negative`)
-		} else if d.IsZero() {
-			return flux.Duration{}, false, errors.New(codes.Invalid, `parameter "every" must be nonzero`)
+			return flux.Duration{}, errors.New(codes.Invalid, `parameter "every" must be non-negative`)
 		}
-		return d, true, nil
-	}()
-	if err != nil {
+		return d, nil
+	}(); err != nil {
 		const docURL = "https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/built-in/transformations/window/#every"
 		return nil, errors.WithDocURL(err, docURL)
-	} else if everySet {
+	} else {
 		spec.Every = every
 	}
 
-	period, periodSet, err := args.GetDuration("period")
-	if err != nil {
+	if period, err := args.GetRequiredDuration("period"); err != nil {
 		const docURL = "https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/built-in/transformations/window/#period"
 		return nil, errors.WithDocURL(err, docURL)
-	}
-	if periodSet {
+	} else {
 		spec.Period = period
 	}
-	if offset, ok, err := args.GetDuration("offset"); err != nil {
+
+	if offset, err := args.GetRequiredDuration("offset"); err != nil {
 		return nil, err
-	} else if ok {
+	} else {
 		spec.Offset = offset
 	}
 
-	if !everySet && !periodSet {
+	if location, err := args.GetRequiredString("location"); err != nil {
+		return nil, err
+	} else {
+		spec.Location = location
+	}
+
+	if spec.Every.IsZero() && spec.Period.IsZero() {
 		const docURL = "https://v2.docs.influxdata.com/v2.0/reference/flux/stdlib/built-in/transformations/window/"
-		return nil, errors.New(codes.Invalid, `window function requires at least one of "every" or "period" to be set`).
+		return nil, errors.New(codes.Invalid, `window function requires at least one of "every" or "period" to be set and non-zero`).
```

Comment:
Oh, is this why? Because this check would catch a zerod every  in many cases?

Reply:
The window function has a strange behavior. If every is not specified, its default value is period. If period is not specified, its default value is every. It's an error for both of them to be unset.

This error message got changed because now it's valid to say `window(every: 0s, period: 1s)` as that will signal that every should use period as its default. So the error message is changed to also specify that you can't do `window(every: 0s, period: 0s)`.